src/Examples/Flowsheets/power_generation/ngcc/ngcc_soec.ipynb

```diff
@@ -4098,153 +4041,84 @@
+      "  23  2.9858219e+00 7.70e-01 1.68e-01  -3.8 1.77e+03    -  2.91e-01 2.86e-01h  2\n",
+      "  24  2.9853856e+00 7.40e-01 1.99e+00  -3.8 1.11e+03    -  6.30e-01 1.75e-01h  3\n",
+      "  25  2.9843096e+00 7.23e-01 1.91e+00  -3.8 1.32e+05    -  4.19e-02 5.64e-02h  1\n",
+      "  26  2.9854879e+00 3.46e-01 8.48e-01  -3.8 6.94e+01  -7.4 1.00e+00 5.50e-01h  1\n",
+      "Reallocating memory for MA57: lfact (330645)\n",
+      "  27  2.9841580e+00 7.70e-02 3.98e-02  -3.8 1.49e+03    -  1.00e+00 1.00e+00h  1\n",
+      "  28  2.9838627e+00 5.51e-02 3.28e-01  -3.8 1.24e+02    -  1.00e+00 1.00e+00h  1\n",
+      "  29  2.9839335e+00 2.59e-03 4.93e-05  -3.8 3.01e+02    -  1.00e+00 1.00e+00h  1\n",
       "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
-      "  50  2.9844188e+00 6.88e-01 1.80e-02  -5.7 3.39e+06    -  1.89e-03 1.04e-03h  2\n",
-      "  51  2.9823927e+00 8.00e-01 1.04e-02  -5.7 1.36e+03  -9.6 1.00e+00 8.50e-01h  1\n",
-      "  52  2.9816814e+00 8.12e-01 1.02e-02  -5.7 3.14e+05    -  2.06e-02 3.27e-03h  2\n",
-      "  53  2.9811499e+00 8.38e-01 1.25e-02  -5.7 1.36e+05    -  1.10e-01 6.71e-03h  2\n",
-      "  54  2.9807349e+00 1.28e+00 1.21e-01  -5.7 2.02e+04    -  9.63e-01 3.59e-02h  1\n",
-      "  55  2.9803918e+00 1.23e+00 1.04e-01  -5.7 3.64e+04 -10.0 5.18e-02 4.23e-02h  2\n",
-      "  56  2.9800112e+00 4.46e-01 4.80e-02  -5.7 1.27e+03    -  5.69e-01 6.13e-01h  1\n",
-      "  57  2.9792950e+00 4.63e-01 4.19e-02  -5.7 2.84e+04    -  1.30e-01 1.27e-01h  1\n",
-      "  58  2.9796317e+00 1.55e-02 7.05e-01  -5.7 3.24e+02    -  3.52e-01 1.00e+00h  1\n",
-      "  59  2.9795780e+00 6.60e-03 2.70e-01  -5.7 7.86e+02    -  1.00e+00 4.12e-01h  1\n",
-      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
-      "  60  2.9795859e+00 1.38e-05 1.67e-06  -5.7 1.17e+01    -  1.00e+00 1.00e+00h  1\n",
-      "  61  2.9795766e+00 2.21e-05 4.77e+00  -7.0 1.18e+01    -  1.00e+00 7.28e-01h  1\n",
-      "  62  2.9795744e+00 2.94e-07 2.66e-01  -7.0 2.47e+00  -8.7 1.00e+00 1.00e+00f  1\n",
-      "  63  2.9795743e+00 1.20e-09 5.53e-09  -7.0 1.57e-01    -  1.00e+00 1.00e+00h  1\n",
+      "  30  2.9839363e+00 4.99e-06 1.19e-07  -3.8 1.06e+01    -  1.00e+00 1.00e+00h  1\n",
+      "  31  2.9803420e+00 9.49e-01 8.00e+02  -5.7 3.62e+03    -  4.20e-01 9.58e-01h  1\n",
+      "  32  2.9794796e+00 2.15e-01 2.84e+02  -5.7 3.12e+03    -  6.71e-01 1.00e+00h  1\n",
+      "  33  2.9794678e+00 1.76e-01 5.92e-01  -5.7 9.56e+02    -  9.98e-01 1.90e-01h  1\n",
+      "  34  2.9795073e+00 6.47e-02 2.93e-01  -5.7 2.22e+01    -  1.00e+00 6.52e-01h  1\n",
+      "  35  2.9795872e+00 8.87e-04 4.90e-04  -5.7 2.46e+01    -  1.00e+00 1.00e+00h  1\n",
+      "  36  2.9795839e+00 4.26e-05 1.31e-06  -5.7 1.08e+00    -  1.00e+00 1.00e+00h  1\n",
+      "  37  2.9795837e+00 9.07e-08 1.37e-09  -5.7 4.90e-02    -  1.00e+00 1.00e+00h  1\n",
```

Comment:
This solver looks far cleaner, although there was a memory reallocation in there.

Reply:
Yeah, unfortunately I'm still not happy with the robustness of this.  I'm going to mess with it a little more today. I also realized that there may be another very small error where when I optimize the capital cost, I impose an upper bound on water use through the sizing of the water treatment system.  I don't add this bound when I lock in the design.  I'll comment when this is ready to review.